app/controllers/concerns/foreman/controller/parameters/keep_param.rb

```diff
@@ -9,6 +9,8 @@ def keep_param(params, top_level_hash, *keys)
     old_params = keys.inject({}) do |op,(key,val)|
       params[top_level_hash].has_key?(key) ? op.update(key => params[top_level_hash].delete(key)) : op
     end
-    yield.update old_params
+    filtered = yield.update(old_params)
+    params[top_level_hash].update old_params
```

Comment:
Please leave a comment here explaining this restores deleted `params` , I don't think it's very clear at first glance